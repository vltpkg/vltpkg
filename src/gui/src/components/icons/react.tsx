export const React = (props: React.SVGProps<SVGSVGElement>) => {
  return (
    <svg
      width="500"
      height="500"
      viewBox="0 0 500 500"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}>
      <g clip-path="url(#clip0_2967_809)">
        <path
          d="m250.504 284.862c270.046 284.862 285.889 269.029 285.889 249.499c285.889 229.968 270.046 214.136 250.504 214.136c230.961 214.136 215.119 229.968 215.119 249.499c215.119 269.029 230.961 284.862 250.504 284.862z"
          fill="currentColor"
        />
        <path
          d="m250.501 321.96c355.363 321.96 440.371 289.522 440.371 249.509c440.371 209.495 355.363 177.058 250.501 177.058c145.639 177.058 60.6318 209.495 60.6318 249.509c60.6318 289.522 145.639 321.96 250.501 321.96z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.724 285.725c240.155 376.482 310.768 433.837 345.442 413.83c380.116 393.823 365.721 304.031 313.29 213.274c260.859 122.517 190.247 65.1621 155.572 85.1689c120.898 105.176 135.293 194.968 187.724 285.725z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.726 213.277c135.295 304.034 120.9 393.826 155.574 413.833c190.248 433.84 260.861 376.485 313.292 285.728c365.723 194.971 380.118 105.179 345.443 85.172c310.769 65.1652 240.157 122.52 187.726 213.277z"
          stroke="currentColor"
          stroke-width="18.913"
        />
      </g>
      <defs>
        <clipPath id="clip0_2967_809">
          <rect
            width="397"
            height="353"
            fill="currentColor"
            transform="translate(52 73)"
          />
        </clipPath>
      </defs>
    </svg>
  )
}
