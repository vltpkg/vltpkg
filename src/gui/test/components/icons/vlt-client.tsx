import { afterEach, expect, test } from 'vitest'
import { cleanup, render } from '@testing-library/react'
import html from 'diffable-html'
import { VltClient } from '@/components/icons/index.ts'

expect.addSnapshotSerializer({
  serialize: v => html(v),
  test: () => true,
})

afterEach(() => {
  cleanup()
})

test('vlt-client icon render default', () => {
  render(<VltClient />)
  expect(window.document.body.innerHTML).toMatchSnapshot()
})

test('vlt-client icon render custom class', () => {
  render(<VltClient className="custom-class" />)
  expect(window.document.body.innerHTML).toMatchSnapshot()
})
