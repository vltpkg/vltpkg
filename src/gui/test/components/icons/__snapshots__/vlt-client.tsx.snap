// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vlt-client icon render custom class 1`] = `

<div>
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width="24"
    height="24"
    viewbox="0 0 24 24"
    fill="none"
    stroke="currentColor"
    stroke-width="2"
    stroke-linecap="round"
    stroke-linejoin="round"
    class="lucide lucide-vlt-client lucide-VltClient custom-class"
    aria-hidden="true"
  >
    <path
      d="M11.1424 6.85711C11.1424 6.88482 11.1418 6.91238 11.1405 6.93977C11.1173 7.42818 11.1468 7.96964 11.4925 8.31539L13.1127 9.93559C13.4585 10.2813 13.9999 10.3108 14.4883 10.2876C14.5157 10.2863 14.5433 10.2857 14.571 10.2857C15.5178 10.2857 16.2853 11.0532 16.2853 12C16.2853 12.9467 15.5178 13.7143 14.571 13.7143C14.5433 13.7143 14.5157 13.7136 14.4883 13.7123C13.9999 13.6891 13.4585 13.7186 13.1127 14.0643L11.4925 15.6845C11.1468 16.0303 11.1173 16.5718 11.1405 17.0602C11.1418 17.0876 11.1424 17.1151 11.1424 17.1428C11.1424 18.0896 10.3749 18.8571 9.42815 18.8571C8.48138 18.8571 7.71387 18.0896 7.71387 17.1428C7.71387 16.196 8.48138 15.4285 9.42815 15.4285C9.45587 15.4285 9.48343 15.4292 9.51082 15.4305C9.99923 15.4537 10.5407 15.4242 10.8864 15.0784L12.5066 13.4583C12.8524 13.1125 12.8819 12.571 12.8587 12.0826C12.8574 12.0552 12.8567 12.0277 12.8567 12C12.8567 11.9723 12.8574 11.9447 12.8587 11.9173C12.8819 11.4289 12.8524 10.8874 12.5066 10.5417L10.8864 8.92148C10.5407 8.57574 9.99923 8.54625 9.51082 8.56944C9.48343 8.57074 9.45587 8.57139 9.42815 8.57139C8.48138 8.57139 7.71387 7.80388 7.71387 6.85711C7.71387 5.91033 8.48138 5.14282 9.42815 5.14282C10.3749 5.14282 11.1424 5.91033 11.1424 6.85711Z"
      fill="currentColor"
      stroke-width="0"
    >
    </path>
  </svg>
</div>

`;

exports[`vlt-client icon render default 1`] = `

<div>
  <svg
    xmlns="http://www.w3.org/2000/svg"
    width="24"
    height="24"
    viewbox="0 0 24 24"
    fill="none"
    stroke="currentColor"
    stroke-width="2"
    stroke-linecap="round"
    stroke-linejoin="round"
    class="lucide lucide-vlt-client lucide-VltClient"
    aria-hidden="true"
  >
    <path
      d="M11.1424 6.85711C11.1424 6.88482 11.1418 6.91238 11.1405 6.93977C11.1173 7.42818 11.1468 7.96964 11.4925 8.31539L13.1127 9.93559C13.4585 10.2813 13.9999 10.3108 14.4883 10.2876C14.5157 10.2863 14.5433 10.2857 14.571 10.2857C15.5178 10.2857 16.2853 11.0532 16.2853 12C16.2853 12.9467 15.5178 13.7143 14.571 13.7143C14.5433 13.7143 14.5157 13.7136 14.4883 13.7123C13.9999 13.6891 13.4585 13.7186 13.1127 14.0643L11.4925 15.6845C11.1468 16.0303 11.1173 16.5718 11.1405 17.0602C11.1418 17.0876 11.1424 17.1151 11.1424 17.1428C11.1424 18.0896 10.3749 18.8571 9.42815 18.8571C8.48138 18.8571 7.71387 18.0896 7.71387 17.1428C7.71387 16.196 8.48138 15.4285 9.42815 15.4285C9.45587 15.4285 9.48343 15.4292 9.51082 15.4305C9.99923 15.4537 10.5407 15.4242 10.8864 15.0784L12.5066 13.4583C12.8524 13.1125 12.8819 12.571 12.8587 12.0826C12.8574 12.0552 12.8567 12.0277 12.8567 12C12.8567 11.9723 12.8574 11.9447 12.8587 11.9173C12.8819 11.4289 12.8524 10.8874 12.5066 10.5417L10.8864 8.92148C10.5407 8.57574 9.99923 8.54625 9.51082 8.56944C9.48343 8.57074 9.45587 8.57139 9.42815 8.57139C8.48138 8.57139 7.71387 7.80388 7.71387 6.85711C7.71387 5.91033 8.48138 5.14282 9.42815 5.14282C10.3749 5.14282 11.1424 5.91033 11.1424 6.85711Z"
      fill="currentColor"
      stroke-width="0"
    >
    </path>
  </svg>
</div>

`;
